repl/src/main/scala/ammonite/repl/interp/Interpreter.scala

```diff
@@ -146,17 +146,24 @@ class Interpreter(prompt0: Ref[String],
       def ivy(coordinates: (String, String, String), verbose: Boolean = true): Unit = {
         val (groupId, artifactId, version) = coordinates
         storage().ivyCache().get((groupId, artifactId, version)) match{
-          case Some(ps) => ps.map(new java.io.File(_)).map(handleJar)
+          case Some(ps) => {
+            ps.map(new java.io.File(_)).map(handleJar)
+            Console.err.println("success")
+          }
           case None =>
-            val resolved = IvyThing.resolveArtifact(groupId, artifactId, version, if (verbose) 2 else 1)
-            storage().ivyCache() = storage().ivyCache().updated(
-              (groupId, artifactId, version),
-              resolved.map(_.getAbsolutePath).toSet
-            )
-
-            resolved.map(handleJar)
+            val resolution = IvyThing.resolveArtifact(groupId, artifactId, version, if (verbose) 2 else 1)
+            resolution map { resolved =>
+              storage().ivyCache() = storage().ivyCache().updated(
+                (groupId, artifactId, version),
+                resolved.map(_.getAbsolutePath).toSet
+              )
+
+              resolved.map(handleJar)
+            } match {
+              case Some(_) => Console.err.println("success")
+              case None => Console.err.println("failure")
+            }
```

Comment:
Maybe consider using `.foreach` instead of `.map` here, since the code is side-effecting and the return value is not needed. For the bottom `match`, I think you can use `resolution` instead.

You could also simplify the output code by doing something like:

```scala
Console.err.println(resolution.fold("failure")(_ => "success"))
```